### PR TITLE
Update locks on case details page

### PR DESCRIPTION
**Pull request by monotypical**
This PR updates the locks on the case details page to match the design, and adds a "view only" badge when no locks are held

Screenshot 2023-06-29 at 11 04 45

### CODE REVIEW

File: cypress/e2e/court-cases/court-case-details/CourtCaseDetails.cy.ts [hunk 1]
```diff
@@ -76,7 +76,6 @@ describe("Court case details", () => {
 
     cy.contains("Case00000")
     cy.contains("Magistrates' Courts Essex Basildon")
-    cy.contains("Urgent")
```
Comment (monotypical) [inline cypress/e2e/court-cases/court-case-details/CourtCaseDetails.cy.ts:79]: Moved this test to `cypress/e2e/court-cases/court-case-details/Header.cy.ts`

File: cypress/e2e/court-cases/court-case-list/CourtCaseList.cy.ts [hunk 2]
```diff
@@ -41,9 +41,12 @@ describe("Court case details", () => {
     cy.visit("/bichard")
 
     cy.contains("a", "NAME Defendant").click()
-    cy.findByText("Case locked by another user").should("not.exist")
-    cy.contains("p", "Trigger locked by: Bichard02").should("exist")
-    cy.contains("p", "Error locked by: Bichard02").should("exist")
+
+    cy.get(".view-only-badge").should("not.exist")
```
Comment (monotypical) [inline cypress/e2e/court-cases/court-case-list/CourtCaseList.cy.ts:45]: Updated these tests to use new format of locks on case details page

File: src/components/Badge.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import ConditionalRender from "components/ConditionalRender"
+
+interface BadgeProps {
+  isRendered: boolean
+  className?: string
+  colour: string
+  label: string
+}
+
+const Badge = ({ isRendered, className, colour, label }: BadgeProps) => {
```
Comment (richardcane) [inline src/components/Badge.tsx:10]: This is probably quite a nice pattern in general for things that need showing and hiding based on roles etc

**Review by richardcane — APPROVED** [review]
All good!

File: src/services/entities/User.ts [hunk 4]
```diff
@@ -39,14 +39,20 @@ export default class User extends BaseEntity {
...
+        group === UserGroup.Allocator ||
+        group === UserGroup.Supervisor
     )
   }
 
   get canLockExceptions() {
     return this.groups.some(
       (group) =>
-        group === "ExceptionHandler" || group === "GeneralHandler" || group === "Allocator" || group === "Supervisor"
+        group === UserGroup.ExceptionHandler ||
```
[the remark this answers is not in the document]
Comment (monotypical) [reply src/services/entities/User.ts]: good catch, I've replaced the functions on the user entity with the ones from `utils/userPermissions` now. I tried to get the user entity to call the other ones but that causes an import cycle as the utils file depends on `User`

**Review by richardcane — APPROVED** [review]

**Pull request merged**